base/shell/explorer/traywnd.cpp

```diff
@@ -3213,7 +3237,13 @@ class CTrayWindowCtxMenu :
                          UINT idCmdLast,
                          UINT uFlags)
     {
-        HMENU menubase = LoadPopupMenu(hExplorerInstance, MAKEINTRESOURCEW(IDM_TRAYWND));
+        HMENU menubase;
+
+        if (g_MinimizedAll.GetSize() == 0 || ::IsThereAnyEffectiveWindow(TRUE))
+            menubase = LoadPopupMenu(hExplorerInstance, MAKEINTRESOURCEW(IDM_TRAYWND));
+        else
+            menubase = LoadPopupMenu(hExplorerInstance, MAKEINTRESOURCEW(IDM_TRAYWND2));
```

Comment:
What's the exact difference(s) between IDM_TRAYWND and IDM_TRAYWND2 menus?
Because, if the only difference is just one different item, then it exists menu APIs that allow you to change the string of an item (that you usually refer by command ID) to reflect the current state.